util/ksonnet/ksonnet.go

```diff
@@ -158,18 +162,39 @@ func (k *ksonnetApp) ListParams() ([]*v1alpha1.KsonnetParameter, error) {
 	if err := json.Unmarshal([]byte(out), &ksParams); err != nil {
 		return nil, err
 	}
-	var params []*v1alpha1.KsonnetParameter
+	params := make(map[string]*v1alpha1.KsonnetParameter)
 	for _, ksParam := range ksParams.Data {
 		value := strings.Trim(ksParam.Value, `'"`)
-		params = append(params, &v1alpha1.KsonnetParameter{
+		params[fmt.Sprintf("%s/%s", ksParam.Component, ksParam.Key)] = &v1alpha1.KsonnetParameter{
 			Component: ksParam.Component,
 			Name:      ksParam.Key,
 			Value:     value,
-		})
+		}
 	}
 	return params, nil
 }
 
+func (k *ksonnetApp) ListParams(environment string) ([]*v1alpha1.KsonnetParameter, error) {
+	params, err := k.listParams("")
```

Comment:
I don't believe it is necessary to list params first without environment specified, and then list params with environment. Ksonnet will give a complete list. Did you find a reason you needed to do this?

Reply:
fixed